src/java.desktop/share/classes/javax/swing/border/BevelBorder.java

```diff
@@ -86,6 +86,10 @@ public BevelBorder(int bevelType) {
     /**
      * Creates a bevel border with the specified type, highlight and
      * shadow colors.
+     * highlightOuterColor and highlightInnerColor will be derived from
```

Comment:
I think we should use just a brief text for "highlightOuterColor"/etc like in the other parts of class "the outer shadow"/"the inner shadow"/"the inner highlight" etc.  Since in this particular case the highlightOuterColor is not a parameter nor a field.
And please update the copyright dates.

Reply:
I guess I have done what is being asked. Are there any other comments? If not, can this PR and CSR be approved please?

Reply:
So as per the updated version of the specification - what could be checked is that the resulting inner/outer/highlight/shadow colors should be either:
 - equal to the passed
 - ::brighter() than the passed
 - darker() than the passed
 ?

Reply:
Yes, the passed highlight and shadow color to the constructor would be equal to getHighlightInnerColor(), getShadowOuterColor but again this is for our jdk implementation. As told in the JBS, these are implementation details which should hold true if JDK's BevelBorder class is in use 
but some other JDK can override the class and can give its own implementation as the class is not 'final'.